src/fmllauncher/java/net/minecraftforge/fml/loading/progress/EarlyProgressVisualization.java

```diff
@@ -30,9 +30,8 @@
 
     public void accept(final Dist dist) {
         if (visualization != null) return;
-        // We don't show the window on Mac because mac is super mega triple shit and can't handle anything out of the ordinary
         final boolean ismac = System.getProperty("os.name").toLowerCase(Locale.ROOT).contains("mac");
-        visualization = dist.isClient() && !ismac && Boolean.parseBoolean(System.getProperty("fml.earlyprogresswindow", "true")) ? new ClientVisualization() : new NoVisualization();
+        visualization = new ClientVisualization(ismac);
```

Comment:
This still needs the disable flag, because there is some edgecases like macs in this case that have issues. So there needs to be a way to work around it.

Plus the entire point of this being off thread is to not block vanilla loading... Having the GL context be created in the main thread has no side effects to Minecraft crating its own GL context in the same thread?

Reply:
I removed the NoVisualization default/backup as I didn't see a point of keeping a dead and empty class, but that could be added back in if you believe it should still be there. That flag is used to switch between glfwPollEvents being called on the secondary thread or it not being called at all and on which thread the window and context are created on (as Win32 requires polling to be on the same thread that it was created on). If glfwPollEvents could be called during the loading then these two paths could be unified under the glfw's defined behavior.

There is no additional blocking going on from these commits, all that is moved to the main thread is the creation and deletion of the context, previously deletion was waited upon anyway so only creation is new work/wait time for the main thread. Actual rendering and context binding still always takes place on a second thread.

In regards to the GL context creation, tldr: no. The OpenGL specification describes behavior with multiple contexts and shared contexts, and in this case the entire context and glfw are cleaned up before Minecraft creates it's context. With GLFW it is actually undefined behavior to create an OpenGL context on any thread _but_ the main thread.

Reply:
oh, i see the disable flag you are talking about, i didnt even realize that was there, let me correct that.

Reply:
disable flag added back in.